pxtblocks/blocklyloader.ts

```diff
@@ -2269,6 +2273,268 @@ namespace pxt.blocks {
+
+        Blockly.Blocks['procedures_defnoreturn'].init = function () {
+            let nameField = new Blockly.FieldTextInput('',
+                (Blockly as any).Procedures.rename);
+            //nameField.setSpellcheck(false); //TODO
+            this.appendDummyInput()
+                .appendField((Blockly as any).Msg.PROCEDURES_DEFNORETURN_TITLE)
+                .appendField(nameField, 'NAME')
+                .appendField('', 'PARAMS');
+            this.setColour(getNamespaceColor('functions'));
+            this.setTooltip((Blockly as any).Msg.PROCEDURES_DEFNORETURN_TOOLTIP);
+            this.setHelpUrl((Blockly as any).Msg.PROCEDURES_DEFNORETURN_HELPURL);
+            this.arguments_ = [];
+            this.setStatements_(true);
+            this.statementConnection_ = null;
+        };
+        installBuiltinHelpInfo(proceduresDefId);
+
+        // builtin procedures_defnoreturn
+        const proceduresCallId = "procedures_callnoreturn";
+        const proceduresCallDef = pxt.blocks.getBlockDefinition(proceduresCallId);
+        const proceduresCallTooltips = <Map<string>>proceduresDef.tooltip;
+
+        Blockly.Blocks['procedures_callnoreturn'] = {
+            init: function () {
+                let nameField = new pxtblockly.FieldProcedure('');
+                nameField.setSourceBlock(this);
+                this.appendDummyInput('TOPROW')
+                    .appendField(proceduresCallDef.block['PROCEDURES_CALLNORETURN_TITLE'])
+                    .appendField(nameField, 'NAME');
+                this.setPreviousStatement(true);
+                this.setNextStatement(true);
+                this.setColour(getNamespaceColor('functions'));
+                // Tooltip is set in renameProcedure.
+                this.setHelpUrl((Blockly as any).Msg.PROCEDURES_CALLNORETURN_HELPURL);
+                this.arguments_ = [];
+                this.quarkConnections_ = {};
+                this.quarkIds_ = null;
+            },
+            /**
+             * Returns the name of the procedure this block calls.
+             * @return {string} Procedure name.
+             * @this Blockly.Block
+             */
+            getProcedureCall: function() {
+                // The NAME field is guaranteed to exist, null will never be returned.
+                return /** @type {string} */ (this.getFieldValue('NAME'));
+            },
+            /**
+             * Notification that a procedure is renaming.
+             * If the name matches this block's procedure, rename it.
+             * @param {string} oldName Previous name of procedure.
+             * @param {string} newName Renamed procedure.
+             * @this Blockly.Block
+             */
+            renameProcedure: function(oldName: string, newName: string) {
+                if (Blockly.Names.equals(oldName, this.getProcedureCall())) {
+                    this.setFieldValue(newName, 'NAME');
```

Comment:
newName is already validated to be not null and legal by blockly before entering this function. 
see: https://github.com/Microsoft/pxt-blockly/blob/master/core/procedures.js#L144